Minutes 2021-06-30 (#235)
solid/authorization-panel: meetings/2021-06-30.md

But it's still assuming that we're linked to a resource.

Henry: It's not that far from WAC since it has access to class for classes of resources that you can describe. Say read access to the class of all resources that are tagged over 18 or tagged as written by a group of people.

Elf: https://github.com/solid/web-access-control-spec#aclaccesstoclass
Just to clarify, that would be WAC+ because WAC doesn't support it.

Henry: Of course, but it's there in the ontology so it could be used and because it's uncommonly used, it's harder to implement.
The accessToClass just needs to be used. That's what you're describing.

Elf: Just saying it's not supported by design in https://github.com/solid/web-access-control-spec

Henry: The spec at the moment is extremely limited but it's not like we don't have to improve it.
Let's see whether things get blocked or we can slowly move forward.
Justin speaks of defining sets of resources and that solution fits very nicely.

Justin: The way it's designed, clients should be able to read and write it on common data structures.

> In the current implementation that we've done, we look at data grants and then know how to apply the permissions on resources depending on the structures of resources and data. The authorization agent that understands data structures knows how to apply permissions.

Reviewer: ```suggestion
In the current implementation that we've done, we look at data grants and then know how to apply the permissions on resources, depending on the structures of resources and data. The authorization agent that understands data structures knows how to apply permissions.
```

The authorization server who has access could look at the grant directly and be able to say whether you have access or not.
Sort of leapfrogging having acp policies applying data grants.

> Elf: If you have an algorithm translating the data grants could even not be required.

Reviewer: ```suggestion
Elf: If you have an algorithm, translating the data grants could even not be required.
```

If using data grants, you need to be warry of keeping a single source of truth.

Henry: You're bringing a whole bunch of architecture changes. I think it's a little bit like where are the grants... We'll have to look at it and see whether it's better than WAC.

> There's a whole thing about why we may want ACP and a lot of the assumptions for justifying it might have been wrong.

Reviewer: ```suggestion
There's a whole thing about why we may want ACP, and a lot of the assumptions for justifying it might have been wrong.
```

We should move on to use cases and there, the precise explanation of how things are going, it's difficult to say just by listening to the argument.

Elf: I'm a little worried that the use cases started as WAC UCRs. But Henry, you're right we should compare the solutions. I think it's important to keep things simple from a application development perspective.

> It's a big change of assumptions to have any application have to understand control rules, because I don't think anything that isn't dedicated shouldn't have to care about that.

Reviewer: ```suggestion
It's a big change of assumptions to have any application have to understand control rules, because I don't think anything that isn't dedicated should have to care about that.
```

> Henry: I see an authz agent which is on the server and the one on the client. Because the use cases are too close, (two servers owned by the same person), if you just had a server with access it knows everything.

Reviewer: ```suggestion
Henry: I see an authz agent which is on the server, and the one on the client. Because the use cases are too close (two servers owned by the same person), if you just had a server with access, it knows everything.
```

That's part of what I address in my pull request which is next.

### PR 232 on PR 231: Initial use case for consent workflow

> Henry: You have distinction between guards on the server and clients. There should be no entities that has overview of all that, but there's one client that has access to all your verifiable credentials, all your claims and that lives on your phone or computer. The wallet agent is guarding/authorising an app to receive a token. It can decide what apps are allowed to do.

Reviewer: ```suggestion
Henry: You have distinction between guards on the server and clients. There should be no entity that has overview of all that, but there's one client that has access to all your verifiable credentials, all your claims; and that lives on your phone or computer. The wallet agent is guarding/authorising an app to receive a token. It can decide what apps are allowed to do.
```

I think you can even use the same vocabulary.
Self sovereign identity point of view.

Elf: I think it's useful to think of things in terms of GNAP spec. Resource owner and requesting party authz agents have different responsibilities.
One is about sharing access with others as a resource owner.

> The other about letting requesting parties doing exactly what they're allowed to do.

Reviewer: ```suggestion
The other about letting requesting parties do exactly what they're allowed to do.
```

Henry: So we have 2 authz agents. You can always delegate. If you delegate too much, you lose sovereignty. I think distinguishing the client/resource authz agents will help.
The client authenticates and knows when it wants to.
The server does the reverse.

Elf: That's what we've been discussing with UMA and claims gathering.
Specific claims gathering can be automated.

> Henry: Perhaps we can commit these and try out and find out in details what UMA is.
> We believe in linked data because it's the web architecture friendly way of doing things.

Reviewer: ```suggestion
Henry: Perhaps we can commit these, and try out and find out in detail what UMA is.
We believe in linked data, because it's the web-architecture–friendly way of doing things.
```

The whole message base thing is different.
The best way is to compare through a use case.
Perhaps there is something missing.

> We need to grow the space of the problem because in the current UC, there is too much potential assumptions of known/shared information.

Reviewer: ```suggestion
We need to grow the space of the problem, because, in the current UC, there is too much potential assumptions of known/shared information.
```

Starting from two pods owned by the same entity is not open enough.

Matthieu: I don't understand why it's named UC1, will this go into UCR document?
...: Current use cases seem to be overlapping in requirements. 

> Elf: First we move the use case to the UCR document right away.
> Two, we do two versions of the use case one simple and one extended version.
> Justin can show a solution to the basic and Henry can work on the extended.

Reviewer: ```suggestion
Elf: First, we move the use case to the UCR document right away.
Second, we do two versions of the use case: one simple, and one extended version.
Justin can show a solution to the basic, and Henry can work on the extended.
```

Henry: If consent is a new technology, we need to prove that solutions can scale.

> Elf: We should go at it progressively and allow everyone to work on the level they want.

Reviewer: ```suggestion
Elf: We should go at it progressively, and allow everyone to work on the level they want.
```

Henry: There could be two parts to the story.

Elf: Henry, could you amend your PR and put the extended version under it?

## Actions

> * Henry to ammend PR 232 to make it a two part story with base and extended case

Reviewer: ```suggestion
* Henry to amend [PR 232](https://github.com/solid/authorization-panel/pull/232) to make it a two part story with base and extended case
```
